src/Cli/dotnet/Parser.cs

```diff
@@ -134,63 +158,16 @@ public static Command GetBuiltInCommand(string commandName)
-            // TODO:CH - we want this for dotnet-new argument reporting, but 
-            //           adding this makes forwarding commands (which can't know
-            //           all of the parameters of their wrapped command by design)
-            //           error. so `dotnet msbuild /t:thing` throws a parse error.
-            // .UseParseErrorReporting(127)
-            .UseParseErrorReporting("new")
-            .UseHelp()
-            .UseHelpBuilder(context => DotnetHelpBuilder.Instance.Value)
-            .UseLocalizationResources(new CommandLineValidationMessages())
-            .UseParseDirective()
-            .UseSuggestDirective()
-            .DisablePosixBinding()
-            .UseTokenReplacer(TokenPerLine)
-            .Build();
-
-        private static CommandLineBuilder UseParseErrorReporting(this CommandLineBuilder builder, string commandName)
+        public static CliConfiguration Instance { get; } = new(ConfigureCommandLine(RootCommand))
         {
-            builder.AddMiddleware(async (context, next) =>
-            {
-                CommandResult currentCommandResult = context.ParseResult.CommandResult;
-                while (currentCommandResult != null && currentCommandResult.Command.Name != commandName)
-                {
-                    currentCommandResult = currentCommandResult.Parent as CommandResult;
-                }
-
-                if (currentCommandResult == null || !context.ParseResult.Errors.Any())
-                {
-                    //different command was launched or no errors
-                    await next(context).ConfigureAwait(false);
-                }
-                else 
-                {
-                    context.ExitCode = 127; //parse error
-                    //TODO: discuss to make coloring extensions public
-                    //context.Console.ResetTerminalForegroundColor();
-                    //context.Console.SetTerminalForegroundRed();
-                    foreach (var error in context.ParseResult.Errors)
-                    {
-                        context.Console.Error.WriteLine(error.Message);
-                    }
-                    context.Console.Error.WriteLine();
-                    //context.Console.ResetTerminalForegroundColor();
-                    var output = context.Console.Out.CreateTextWriter();
-                    var helpContext = new HelpContext(context.HelpBuilder,
-                                                      context.ParseResult.CommandResult.Command,
-                                                      output,
-                                                      context.ParseResult);
-                    context.HelpBuilder
-                           .Write(helpContext);
-                }
-            }, MiddlewareOrder.ErrorReporting);
-            return builder;
-        }
+            EnableDefaultExceptionHandler = false,
+            EnableParseErrorReporting = true,
```

Comment:
before we only used Parse Error Reporting for `dotnet new` because the other commands that forward arguments were somehow parse errors. is this fixed now? did we inadvertently fix this when going through and marking more commands with `TreatUnparsedTokensAsError = false`?

Reply:
Nevermind - I see in Program.cs where this is handled more explicitly.